_posts/2018-10-29-go.md

```diff
@@ -0,0 +1,41 @@
+---
+layout: post
+title: Go over Nabla- App Safety meets Host Isolation
+---
+
+This blog presents how a Go application can be built and run as a Nabla container.
+We take an example of a web server application, part of the [nabla-demo-apps](https://github.com/Nabla-containers/nabla-demo-apps) repository.
+To enable/port Go over Nabla, we also briefly present certain key modifications made to the Go runtime.
+
+
+
+### Application build
+The httpd example highlights Nabla's support for Go applications organized across multiple .go files, multiple packages, as well as third-party package imports from github.
+
+In the [sample code](https://github.com/nabla-containers/nabla-demo-apps/tree/master/go-httpd/src), the `main()` function in httpd.go uses a utility function from the helper package (helper/process_args.go) to parse the commandline arguments. 
+After some trivial commandline parsing, the webserver awaits HTTP GETs at port 3000 from the outside world.
+
+Building this Go webserver as a Nabla container is pretty straightforward. 
+[Dockerfile.Nabla](https://github.com/nabla-containers/nabla-demo-apps/blob/master/go-httpd/Dockerfile.nabla) lists the steps required for a multi-stage build of the Go app.
+First, the app's source directory is added to the [`nabla-go-base`](https://github.com/nabla-containers/nabla-base-build/tree/master/go-base) image, which contains the Go runtime ported over Nabla (see Section 'Under the Covers'). 
+This is followed by the usual `go get` of the dependencies.
+After this, a generic [Makefile.Goapp](https://github.com/nabla-containers/nabla-base-build/blob/master/go-base/Makefile.goapp) (part of the base image itself) is triggered, which builds the Go app as a static Nabla binary.
+The resulting binary then becomes the entrypoint of the Go app Nabla container, which can be run using the [`runnc`](https://github.com/nabla-containers/runnc) Nabla-container runtime as `docker run -it --rm --runtime=runnc go-httpd-Nabla`.
+
+
+### Under the covers
```

Comment:
I think it will be nice to let the reader know that the following content is rated R for rumprun, so some details may make more sense to those who are familiar with rump and want to dig deeper.